modules/core/src/main/scala/dev/profunktor/redis4cats/effect/RedisEc.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2018-2020 ProfunKtor
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.profunktor.redis4cats.effect
+
+import cats.effect._
+import java.util.concurrent.Executors
+
+trait RedisEc {
+  def delay[F[_]: Sync: ContextShift, A](thunk: => A): F[A]
+  def eval[F[_]: ContextShift, A](fa: F[A]): F[A]
+}
```

Comment:
I guess we'd want to make this extend `AnyVal`, which would require inlining the implementation since `Blocker` also extends `AnyVal`

Reply:
How about this?

```scala
trait RedisExecutor[F[_]] {
  def delay[A](thunk: => A): F[A]
  def eval[A](fa: F[A]): F[A]
}

object RedisExecutor {
  def make[F[_]: ContextShift: Sync]: Resource[F, RedisExecutor[F]] =
    Blocker.fromExecutorService(F.delay(Executors.newFixedThreadPool(1))).map { ec => 
      new RedisExecutor[F] {
        def delay[A](thunk: => A): F[A] = ec.delay(thunk)
        def eval[A](fa: F[A]): F[A] = ec.blockOn(fa
      }
    } 
}
```

Reply:
That works. I wonder if we should make `RedisExecutor` and the implementation classes that use it package-private?

Reply:
Yes, that would be great :+1:

Reply:
Cool, then it's also safe to pass the RedisExecutor implicitly everywhere 👍